src/jsr.rs

```diff
@@ -192,7 +192,7 @@ impl<'de> Deserialize<'de> for JsrDepPackageReq {
     D: serde::Deserializer<'de>,
   {
     let text = String::deserialize(deserializer)?;
-    match Self::from_str(&text) {
+    match Self::from_str_loose(&text) {
```

Comment:
For serializing, we should parse it loosely in order to support all the different constraints that are supported in a package.json.

Reply:
Robustness principle at play here 😎